lib/rspec/core/reporter.rb

```diff
@@ -79,6 +88,17 @@ def message(message)
       notify :message, Notifications::MessageNotification.new(message)
     end
 
+    # @param event [Symbol] Name of the custom event to trigger on formatters
+    # @param options [Hash] Hash of arguments to provide via `CustomNotification`
+    #
+    # Publish a custom event to supporting registered formatters.
+    # @see RSpec::Core::Notifications::CustomNotification
+    def publish(event, options={})
+      event_name = event.to_sym
```

Comment:
This has the side effect of supporting `event` as a string (in spite of the method signature stating `event` should be a `Symbol`), which makes this API more tolerant of input args than I'd like it to be.  Anyhow, I looked at the implementation of `notify` and it looks like String event names will never notify any listeners anyway.

* When listeners are [registered](https://github.com/rspec/rspec-core/blob/v3.2.0/lib/rspec/core/reporter.rb#L33) it converts their supported events to symbols and uses a symbol as the key in the `@listeners` hash.
* [notify](https://github.com/rspec/rspec-core/blob/v3.2.0/lib/rspec/core/reporter.rb#L145) calls `registered_listeners(event)` to get the list of listeners, with no coercion of the `event` arg.
* [registered_listeners](https://github.com/rspec/rspec-core/blob/v3.2.0/lib/rspec/core/reporter.rb#L39-L41) uses the provided argument as-is (no `to_sym` conversion) to lookup the listeners in the hash.
* Therefore, `notify` is always a no-op when passed an argument that's not a symbol, since there will never be any listeners.

So I'd say you can remove this because a string event name won't send the notification to any formatters, anyway.  That said, having `publish` prevent notification of lifecycle events in a string form is a very useful guarantee to have, and relying on the current implementation of the methods I listed above is quite implicit.  I'd suggest encoding this requirement in a test that tries using `publish` with a string version of a lifecycle event and expects no formatters to be notified -- that way, if future changes to the above listed methods started allowing string events to be notified the spec would fail and we could revisit doing some kind of more explicit prevention here.